CONTRIBUTING.md

```diff
@@ -2,149 +2,152 @@
+   cd tidb-nightly-darwin-amd64
+   ```
 
-Download all needed files, but do not save them in `tidb-dashboard` folder, Choose 1 of 2:
+2. Start a PD server
 
-For linux user:
-```bash
-wget https://download.pingcap.org/tidb-latest-linux-amd64.tar.gz
-tar -xzf tidb-latest-linux-amd64.tar.gz
-cd tidb-latest-linux-amd64
-```
+   ```bash
+   ./bin/pd-server --name=pd --data-dir=pd --client-urls=http://127.0.0.1:2379 --log-file=pd.log
+   # Now pd-server is listen on port 2379
+   ```
 
-For Mac OS user: 
-```bash
-wget https://download.pingcap.org/tidb-nightly-darwin-amd64.tar.gz
-wget https://download.pingcap.org/tikv-nightly-darwin-amd64.tar.gz
-wget https://download.pingcap.org/pd-nightly-darwin-amd64.tar.gz
-mkdir tidb-nightly-darwin-amd64
-tar -xzf tidb-nightly-darwin-amd64.tar.gz -C tidb-nightly-darwin-amd64 --strip-components=1
-tar -xzf tikv-nightly-darwin-amd64.tar.gz -C tidb-nightly-darwin-amd64 --strip-components=1
-tar -xzf pd-nightly-darwin-amd64.tar.gz -C tidb-nightly-darwin-amd64 --strip-components=1
-cd tidb-nightly-darwin-amd64
-```
+3. Start a TiKV server
 
-Then start a local TiDB cluster with 1 TiDB, 1 TiKV and 1 PD step by step:
+   Open a new terminal:
 
-Firstly start a pd-server:
+   ```bash
+   ./bin/tikv-server --addr="127.0.0.1:20160" --pd-endpoints="127.0.0.1:2379" --data-dir=tikv --log-file=./tikv.log
+   # Now tikv-server is listen on port 20160
+   ```
 
-```bash
-./bin/pd-server --name=pd --data-dir=pd --client-urls=http://127.0.0.1:2379 --log-file=pd.log
-# Now pd-server is listen on port 2379
-```
+4. Start a TiDB server
 
-Open a new terminal and start tikv-server:
+   Open a new terminal:
 
-```bash
-./bin/tikv-server --addr="127.0.0.1:20160" --pd-endpoints="127.0.0.1:2379" --data-dir=tikv --log-file=./tikv.log
-# Now tikv-server is listen on port 20160
-```
+   ```bash
+   ./bin/tidb-server --store=tikv --path="127.0.0.1:2379" --log-file=tidb.log
+   # Now tidb-server is listen on port 4000
+   ```
 
-Open a new terminal and start tidb-server:
+5. Use mysql-client to check everything works fine:
 
-```bash
-./bin/tidb-server --store=tikv --path="127.0.0.1:2379" --log-file=tidb.log
-# Now tidb-server is listen on port 4000
-```
+   ```bash
+   mysql -h 127.0.0.1 -P 4000 -uroot
+   ```
 
-Use `mysql-client` to check if `tidb-server` is on:
+</details>
 
-```bash
-mysql -h 127.0.0.1 -P 4000 -uroot
-```
+### Step 2. Prepare Prerequisites
 
-### Building and running
+The followings are required for developing TiDB Dashboard:
 
-At this point, you can build and run TiDB Dashboard. 
+- git - Version control
+- make - Build tool (run common workflows)
+- [Golang 1.13+](https://golang.org/) - To compile the server.
+- [Node.js 12+](https://nodejs.org/) - To compile the front-end.
+- [Yarn 1.21+](https://classic.yarnpkg.com/en/docs/install) - To manage front-end dependencies。
+- [Java 8+](https://www.java.com/ES/download/) - To generate JavaScript API client by OpenAPI specification.
```

Comment:
Do we need add [flamegraph](https://github.com/brendangregg/FlameGraph) here?

Reply:
It doesn't affect other features for now. I think we don't need to explicit mention it.